vispy/scene/cameras/base_camera.py

```diff
@@ -465,13 +489,86 @@ def _set_scene_transform(self, tr):
+                                self.last_self_top = self.get_state()["rect"].top
+                                self.last_cam_top = cam.get_state()["rect"].top
+                                self.last_self_bottom = self.get_state()["rect"].bottom
+                                self.last_cam_bottom = cam.get_state()["rect"].bottom
+                elif (self._linked_cameras[0][1] == "y"):
+                    for i in range(len(self._linked_cameras)):
+                        cam = self._linked_cameras[i][0]
+                        self_state = self.get_state()
+                        cam_state = cam.get_state()
+                        s = self_state
+                        for u in range(2):
+                            if(u == 1):
+                                if(self.last_self_pos is not None and self.last_cam_pos is not None and self.last_cam_left is not None and self.last_cam_right is not None  and self.last_self_left is not None and self.last_self_right is not None):
+                                    self_state["rect"].pos = (self.last_cam_pos[0], self_state["rect"].pos[1])
+                                    self_state["rect"].left = self.last_cam_left
+                                    self_state["rect"].right = self.last_cam_right
+                                    cam.set_state(self_state)
+                                    self_state["rect"].pos = (self.last_self_pos[0], self_state["rect"].pos[1])
+                                    self_state["rect"].left = self.last_self_left
+                                    self_state["rect"].right = self.last_self_right
+                                    self.set_state(self_state)
+                                    self.last_self_pos = self.get_state()["rect"].pos
+                                    self.last_cam_pos = cam.get_state()["rect"].pos
+                                else:
+                                    self.last_self_pos = self.get_state()["rect"].pos
+                                    self.cam_self_pos = cam.get_state()["rect"].pos
+                                    self.last_self_left = self.get_state()["rect"].left
+                                    self.last_cam_left = cam.get_state()["rect"].left
+                                    self.last_self_right = self.get_state()["rect"].right
+                                    self.last_cam_right = cam.get_state()["rect"].right
+                            else:
+                                self.last_self_pos = self.get_state()["rect"].pos
+                                self.last_cam_pos = cam.get_state()["rect"].pos
+                                self.last_self_left = self.get_state()["rect"].left
+                                self.last_cam_left = cam.get_state()["rect"].left
+                                self.last_self_right = self.get_state()["rect"].right
+                                self.last_cam_right = cam.get_state()["rect"].right
```

Comment:
This pattern of:

```
if something1:
    if something2:
        # code1
    else:
        # code 2
else:
    # code2
```

Or at least what seems like very similar code could be updated to something more like:

```
if something1 and somethine2:
    # code1
# code2
```

Thoughts?

Reply:
I tried to combine (u == 1) and (x is not None) however in my test script it breaks the link and it doesnt work